numba/npyufunc/__init__.py

```diff
@@ -6,3 +6,22 @@
 if hasattr(_internal, 'PyUFunc_ReorderableNone'):
     PyUFunc_ReorderableNone = _internal.PyUFunc_ReorderableNone
 del _internal, array_exprs
+
+
+def _init():
+
+    def init_vectorize():
+        from numba.cuda.vectorizers import CUDAVectorize
+        return CUDAVectorize
+
+    def init_guvectorize():
+        from numba.cuda.vectorizers import CUDAGUFuncVectorize
+        return CUDAGUFuncVectorize
+
+    Vectorize.target_registry.ondemand['gpu'] = init_vectorize
```

Comment:
Whilst having a target called 'gpu' before probably was OK, I think as this API is new to Numba, we should probably not refer to it as 'gpu', in order to not cause confusion if and when ufuncs on non-CUDA GPU-like architectures are supported. (In other words this line and the other one referencing 'gpu' should probably be removed).